integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/LuceneExtensionIT.java

```diff
@@ -77,27 +87,99 @@
 	public ExpectedException thrown = ExpectedException.none();
 
 	private SearchIntegration integration;
-	private IndxMapping indexMapping;
+	private IndexMapping indexMapping;
 	private StubMappingIndexManager indexManager;
 
 	@Before
 	public void setup() {
 		this.integration = setupHelper.withDefaultConfiguration( BACKEND_NAME )
 				.withIndex(
 						INDEX_NAME,
-						ctx -> this.indexMapping = new IndxMapping( ctx.getSchemaElement() ),
+						ctx -> this.indexMapping = new IndexMapping( ctx.getSchemaElement() ),
 						indexManager -> this.indexManager = indexManager
 				)
 				.setup();
 
 		initData();
 	}
 
+	@Test
+	@SuppressWarnings("unused")
+	public void queryContext() {
+		StubMappingSearchScope scope = indexManager.createSearchScope();
+
+		// Put intermediary contexts into variables to check they have the right type
+		LuceneSearchQueryResultDefinitionContext<DocumentReference, DocumentReference> context1 =
+				scope.query().extension( LuceneExtension.get() );
```

Comment:
The whole point of the extension system is to avoid having to add a dependency from the engine or (mapper-orm) module to the backend modules. Doing what you propose would require to add that dependency. Doing so would mean that someone using the ORM mapper with the Elasticsearch backend would still have a dependency to the Lucene backend, and through it would have a dependency to Lucene, and that's what we've been trying to avoid the whole time.

The extension system is admittedly awkward, but it works. Personally I can't find another solution that wouldn't require to add unwanted dependencies between modules. We can start such a discussion if you have a better idea, but given we already have many extensions throughout the codebase, I'd rather merge this first and address this issue separately (separate ticket, thread on the mailing list, ...).